src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,296 @@
+                Streams = streams,
+            };
+        }
+
+        private async Task<IDictionary<string, IObservable<ExecutionResult>>?> ExecuteSubscriptionNodesAsync(ExecutionContext context, ExecutionNode[] nodes)
+        {
+            var streams = new Dictionary<string, IObservable<ExecutionResult>>();
+
+            foreach (var node in nodes)
+            {
+                var stream = await ResolveEventStreamAsync(context, node).ConfigureAwait(false);
+                if (stream == null)
+                    return null;
+                streams[node.Name!] = stream;
+            }
+
+            return streams;
+        }
+
+        /// <summary>
+        /// Asynchronously returns a stream of <see cref="ExecutionResult"/> responses for the
+        /// specified <see cref="ExecutionNode"/>.
+        /// </summary>
+        protected virtual async Task<IObservable<ExecutionResult>?> ResolveEventStreamAsync(ExecutionContext context, ExecutionNode node)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var resolveContext = new ReadonlyResolveFieldContext(node, context);
+
+            IObservable<object?> subscription;
+
+            try
+            {
+                if (node.FieldDefinition?.Subscriber == null)
+                {
+                    // this is normally caught by schema validation
+                    throw new InvalidOperationException($"Subscriber not set for field '{node.Field.Name}'.");
+                }
+
+                subscription = await node.FieldDefinition.Subscriber.SubscribeAsync(resolveContext).ConfigureAwait(false);
+
+                if (subscription == null)
+                {
+                    throw new InvalidOperationException($"No event stream returned for field '{node.Field.Name}'.");
+                }
+            }
+            catch (OperationCanceledException) when (context.CancellationToken.IsCancellationRequested)
+            {
+                throw;
+            }
+            catch (ExecutionError error)
+            {
+                error.Path = node.ResponsePath;
+                error.AddLocation(node.Field, context.Document);
+                context.Errors.Add(error);
+                return null;
+            }
+            catch (Exception exception)
+            {
+                context.Errors.Add(await HandleExceptionInternalAsync(context, node, exception,
+                    $"Could not subscribe to field '{node.Field.Name}'.").ConfigureAwait(false));
+                return null;
+            }
+
+            // cannot throw an exception here
+            return subscription
+                .SelectCatchAsync(
+                    async (value, token) =>
+                    {
+                        // duplicate context to prevent multiple event streams from sharing the same context,
+                        // clear errors/metrics/extensions, and free array pool leased arrays
```

Comment:
```suggestion
                        // and as a consequence clearing errors/metrics/extensions and freeing array pool leased arrays
```

Did I understand the meaning correctly?

Reply:
well, it's not a consequence, nor does it happen on the original context -- but my comment was inaccurate also.  i'll modify

Reply:
```suggestion
                        // and clear errors/metrics/extensions. Free array pool leased arrays after execution.
```

Reply:
Please review revised comment suggestion.  I'll merge now so tests can pass and all that.

Reply:
OK